Arrowgene.Ddon.GameServer/Characters/CraftManager.cs

```diff
@@ -408,6 +446,13 @@ public static bool CanPawnExpUp(Pawn pawn)
 
         public static void HandlePawnExpUpNtc(GameClient client, Pawn leadPawn, uint exp, uint bonusExp)
         {
+            uint expUp = 0;
+            uint bonusExpUp = 0;
+            if (CanPawnExpUp(leadPawn))
+            {
+                expUp = exp;
+                bonusExpUp = bonusExp;
```

Comment:
If you want you can add support to the GP Course manager to detect usage of `GP_COURSE_EFFECT_PAWN_CRAFT_EXP_UP` to calculate the `bonusExp` value.